tasks/libs/github_actions_tools.py

```diff
@@ -190,11 +191,12 @@ def download_artifacts_with_retry(run_id, destination=".", retry_count=3, retry_
             download_artifacts(run_id, destination)
             print(color_message(f"Successfully downloaded artifacts for run {run_id} to {destination}", "blue"))
             return
-        except ConnectionResetError:
+        except requests.exceptions.RequestException:
             retry -= 1
             print(f'Connectivity issue while downloading the artifact, retrying... {retry} attempts left')
```

Comment:
I guess you should print this message before decreasing `retry` (otherwise you could print 0 attempt left while you will call once more download artifacts)

Reply:
Hmm I think that when 0 attempts left is printed, retry = 0 and we do not call `download_artifacts`. I though about print after or before decreasing the counter, and I felt like it was good like that but maybe I missed it.

Reply:
yes, sorry you are right, no more call is done. What is weird to me it to print that you will do no more calls